app/models/concerns/orchestration/compute.rb

```diff
@@ -124,7 +128,7 @@ def setComputeDetails
       attrs.each do |foreman_attr, fog_attr|
         if foreman_attr == :mac
           return false unless match_macs_to_nics(fog_attr)
-        elsif foreman_attr == :ip
+        elsif [:ip, :ip6].include?(foreman_attr)
           value = vm.send(fog_attr) || find_address
```

Comment:
`find_address` isn't IPv6 aware here, it'll call `vm.ip_addresses` which can return v4 addresses if the compute claims to provide v6 but `vm.send(fog_attr)` returns nil. You can reproduce this by _not_ ticking the IPv6 box when using fog-digitalocean.

It probably needs to be changed to filter addresses from `vm.ip_addresses` depending on their type?